subiquity/models/identity.py

```diff
@@ -14,43 +14,86 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import logging
+from typing import Any, Optional, Type
 
 import attr
 
+from subiquity.common.types import IdentityData
+
 log = logging.getLogger("subiquity.models.identity")
 
 
-@attr.s
-class User(object):
-    realname = attr.ib()
-    username = attr.ib()
-    password = attr.ib()
+class DefaultGroups:
+    """Special value for unresolved default groups"""
+
+
+@attr.s(auto_attribs=True)
+class User:
+    realname: str
+    username: str
+    password: str
+
+    groups: set[str | Type[DefaultGroups]]
+
+    @classmethod
+    def from_identity_data(cls, data: IdentityData) -> "User":
+        return cls(
+            username=data.username,
+            password=data.crypted_password,
+            realname=data.realname if data.realname else data.username,
+            groups={DefaultGroups},
+        )
 
+    @classmethod
+    def from_autoinstall(cls, data: dict[str, Any]) -> "User":
+        if "groups" not in data:
+            groups = {DefaultGroups}
+        elif isinstance(data["groups"], list):
+            groups = set(data["groups"])
+        elif isinstance(data["groups"], dict):
+            if "override" in data["groups"]:
+                groups = set(data["groups"]["override"])
+            elif "append" in data["groups"]:
+                groups = {DefaultGroups}
+                groups.update(set(data["groups"]["append"]))
+            else:
+                raise ValueError
+        else:
+            raise ValueError
+        return cls(
+            username=data["username"],
+            realname=data.get("realname", ""),
```

Comment:
question: if we don't pass a `realname` in the identity flow then it will take the value of the user name. Here we set it to empty. Is the behavior difference intentional?

Reply:
Oh I see this is the original behavior, but maybe more clear now they are closer together. I'll leave the question up so we can consider it but not blocking imo.